app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageViewWrapper.kt

```diff
@@ -17,18 +26,22 @@ enum class MessageStatus(val stringResourceId: Int) {
     Untouched(-1), Deleted(R.string.label_message_status_deleted), Edited(R.string.label_message_status_edited)
 }
 
-data class Message(
-    val user: User,
-    val messageSource: MessageSource = MessageSource.CurrentUser,
-    val messageHeader: MessageHeader,
-    val messageContent: MessageContent,
-) {
-    val isDeleted: Boolean = messageHeader.messageStatus == MessageStatus.Deleted
-}
 
 sealed class MessageContent {
     data class TextMessage(val messageBody: MessageBody) : MessageContent()
-    data class ImageMessage(val imageUrl: String) : MessageContent()
+    data class ImageMessage(val rawImgData: ByteArray?, val width: Int, val height: Int) : MessageContent() {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+            other as ImageMessage
+            if (!rawImgData.contentEquals(other.rawImgData)) return false
+            return true
+        }
+
+        override fun hashCode(): Int {
+            return rawImgData.contentHashCode()
```

Comment:
Just out of curiosity, the `equals` and `hashCode` methods generally when auto-generating, it will consider all the fields values on the object for the comparison, I think to represent real state and diff between instances. Did we get intentionally out of that comparison, the `width` and `height` ? Which would kind of make sense I think

Reply:
Yes, indeed that shouldn't be a problem. The `equals` comparison is basically to allow us to compare the raw content of the image. The `width` and `height` are just metadata that we use to pass through the final `MessageImage` object used to display the inline image in the conversation. So basically, 2 `ImageMessage` objects with the same raw image but different `width` and `height` (although I'd say in practise that would be impossible) would still be considered the same image. 
But good point, we should definitely need to take a look into this the moment we start considering other new features like message forwarding from different conversations, etc...